src/qt/blockview.cpp

```diff
@@ -277,6 +277,28 @@ GuiBlockView::~GuiBlockView()
         m_validation_interface = nullptr;
     }
 }
+void GuiBlockView::changeEvent(QEvent* e)
+{
+    if (e->type() == QEvent::PaletteChange) {
+        // Update existing bubble colors with new theme
+        const QColor bg_colour = palette().color(backgroundRole());
+        const bool dark_mode = GUIUtil::isDarkMode(bg_colour);
+        const QColor color_blue = dark_mode ? QColor(137, 170, 255) : QColor(2, 61, 204);
+        
+        // Update all existing bubbles (need to lock mutex for thread safety)
```

Comment:
Maybe we should store m_bubble_colour on the class (and reuse it when making new bubbles), so we can avoid doing this if it hasn't changed?